tests/test_shared.py

```diff
@@ -154,7 +154,7 @@ def test_state_dict_too_large(local_remote_dir: Tuple[str, str]):
 @pytest.mark.parametrize('dtype', [np.int32, np.int64, np.float32, np.float64])
 @patch('streaming.base.shared.array.SharedMemory')
 def test_shared_array_size_is_integer(mock_shared_memory, dtype):
-    SharedArray(shape=(2, 3), dtype=dtype, name='test_shared_array')
+    SharedArray(3, np.int32, name='test_shared_array')
```

Comment:
this isn't testing the other dtypes, right? I think you could keep the earlier change but include some assert statements before this line so that type checking passes.

Reply:
Nice catch!